### PR TITLE
Fix error propagation from list init

### PR DESCRIPTION
**Pull request by t-vi**
@kiya00 With any bit of luck, this should also unblock your enumerate PR.

### CODE REVIEW

File: thunder/core/interpreter.py [hunk 1]
```diff
@@ -2101,8 +2101,9 @@ def __new__(cls, iterable=()):
         return wrap_const(cls.value())
 
     def __init__(self, iterable=()):
-        SequenceWrapperMethods.__init__(self, iterable)
-        return wrap_const(None)
+        # We need to propagate the return value because it could be JIT_SIGNALS
+        res = SequenceWrapperMethods.__init__(self, iterable)
+        return res
```
Comment (Borda) [inline thunder/core/interpreter.py:2106]:
```suggestion
        return SequenceWrapperMethods.__init__(self, iterable)
```

**Review by nikitaved — APPROVED** [review]
LGTM! Thank you, @t-vi !

File: thunder/core/interpreter.py (hunk 1 above)
Comment (t-vi) [reply thunder/core/interpreter.py:2106]: Yeah, we could, but I exect to add debug prints soonish.

**Comment by t-vi** [comment]
Thank you for reviewing @nikitaved  @Borda

**Pull request merged**